connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -200,48 +235,68 @@ public boolean isSinkConnector(String connName) {
         return result;
     }
 
-    public void stopConnectors() {
-        stopConnectors(new HashSet<>(connectors.keySet()));
+    private void stopConnectors() {
+        stopConnectors(new ArrayList<>(connectors.keySet()));
     }
 
-    public Collection<String> stopConnectors(Collection<String> connectors) {
-        final List<String> stopped = new ArrayList<>(connectors.size());
-        for (String connector: connectors) {
-            if (stopConnector(connector)) {
-                stopped.add(connector);
-            }
-        }
-        return stopped;
+    private void stopConnectors(Collection<String> connectors) {
```

Comment:
Indeed, if we are always using it over the whole set. 

Immutability via copy was kind of inherited code which I didn't want to disrupt at this point. I'll take your word for it.